Makefile

```
@@ -82,6 +91,41 @@ check-coverage:
 	fi && \
 	echo "✅ Coverage $$COV% meets threshold $(COVERAGE_THRESHOLD)%"
 
+# Check coverage for staged production Go files only. This keeps pre-commit focused on
+# files touched by the commit instead of unrelated low-coverage code in the same package.
+check-coverage-staged:
```

checkmake (0.3.2)

[warning] Target body for "check-coverage-staged" exceeds allowed length of 5 lines (31).

(maxbodylength)